### PR TITLE
Refactor: extract TestFlags, TestConfig and DeployableConfig from pilot e2e tests

### PR DESCRIPTION
**Pull request by vadimeisenbergibm**
In order to use them in Bookinfo v1aplpha3 tests

### CODE REVIEW

File: tests/e2e/framework/deployable_config.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+type DeployableConfig struct {
+	Namespace  string
+	YamlFiles  []string
+	applied    []string
+	Kubeconfig string
+}
+
+// Setup pushes the config and waits for it to propagate to all nodes in the cluster.
+func (c *DeployableConfig) Setup() error {
+	c.applied = []string{}
```
Comment (costinm) [inline tests/e2e/framework/deployable_config.go]: Please add an env variable ( or flag, but rather env ) to 'skip_setup' and 'skip_cleanup' - we have a global flag for skipping istio setup as well. 

Will help a lot for debugging individual test cases - you can run a test with the env set, and debug manually after the test fails. Or repeat the test - without the delay.

File: tests/e2e/tests/pilot/pilot_test.go [hunk 2]
```diff
@@ -27,31 +27,28 @@ import (
...
+	httpOK               = "200"
+	ingressAppName       = "ingress"
+	ingressContainerName = "ingress"
 )
 
 var (
-	tc = &testConfig{
+	tc *framework.TestConfig
+	tf = &framework.TestFlags{
 		V1alpha1: true, //implies envoyv1
```
Comment (costinm) [inline tests/e2e/tests/pilot/pilot_test.go]: Please fix the defaults - we never support both true. I think default should be alpha3 - so running without any options (from IDE) will test the new stuff.

**Comment by vadimeisenbergibm** [comment]
@costinm Could you please review this PR and merge?

**Pull request closed without merging**